### DISCUSSION TITLE
How to exclude PATHs in bicycle navigation?

### DISCUSSION THREAD

**Discussion by ebnater01** [osmandapp/OsmAnd#19297] [Q&A]
![osmand](https://github.com/osmandapp/OsmAnd/assets/163061747/b524f861-46bc-4c76-8a18-7a8ba6282c86)

I really like this APP, but yet it doesn't do what I really want... In our woods there are several types of ways. OSM knows that and shows them with different symbol. The green encirceled ways on the attached picture are fine for biking. They are made of fine gravel and really good to drive. The RED encirceled way is a simple path, no gravel or anything else, just wood soil with some trampeled over it. You have stones, sticks, roots,... Yes, you can go there by bike, but you have to really slow down and take care.

OSM knows the difference and shows them with different line types. HOW can I use this information to setup navigation / routing? How to say OSMAND: "don't take that way!!". How to tell it to OSMAND generally to ignore that kind of paths?

Thanks!!

**Comment by DmitryAlexei** [comment]
Hi @ebnater01! 
- About bicycle routing options, you can read in the docs. Here is the link - https://osmand.net/docs/user/navigation/routing/bicycle-based-routing#route-parameters---bicycle
I think, in your case choosing _Avoid footways_ may help.

- Another option is to add this specific road to the list of non-used for routing roads. Try Avoid roads option in the Context menu. For more information see https://osmand.net/docs/user/map/map-context-menu#avoid-road

**Reply by ebnater01** [reply]
Hi @DmitryAlexei !

Thanks for your answer, unfortunately they didn't solve my problem. 

- I tried the bicycle routing options, including _Avoid Footpaths_, but it still takes the "wrong" way. 

- _Avoid roads_ is the workaround I currently already use for the three specific roads I know here in the woods. I know them by experience, since OSMAND led me here and I recognized, that this is not the type of path I want to bike. So this helps for known roads, but not "in advance", when I plan a new route into unknown terrain. 

In the bicycle routing options there was a link to MTB routing. This is the thing I will try next. Perhaps playing with the evaluation options of the different MTB track types helps. I still hope, that there might be an option selectable or a parameter file or something, where I can exclude specific way types, pavements,... - just all the different information, that already is present in OSM.

**Reply by tommycrock** [reply]
It would be helpful to know how those ways are currently tagged.

**Comment by ebnater01** [comment]
@tommycrock main difference is in _tracktype_ . While the green encirceled tracks have "_tracktype   grade2_", the red encirceled bad track has "_tracktype    grade4_". Depending on what tag I look at, there are also other differences between the tracks, but _tracktype_ is consistent for also the other roads I mean.

Is there a way to exclude certain _tracktype_ grades from routing?

**Reply by DmitryAlexei** [reply]
You can try to modify routing.xml file by yourself. You can read how to do that by the link https://osmand.net/docs/user/navigation/routing/osmand-routing#customize-offline-routing

Here is a link to avoid footways for bicycle inside that file [code lines](https://github.com/osmandapp/OsmAnd-resources/blob/864e852e7e98fe588dfe90b5a0560a8856b8e6fa/routing/routing.xml#L1679C4-L1692C69)

**Comment by ebnater01** [comment]
@DmitryAlexei: at first look that is great and what I need! At second look I recognize, that it for sure will take some time and trials, since it looks quite complicated. But I will try at the weekend. **Thanks!!**
